lib/matplotlib/tests/test_legend.py

```diff
@@ -1667,3 +1667,120 @@ def test_boxplot_legend_labels():
     bp4 = axs[3].boxplot(data, label='box A')
     assert bp4['medians'][0].get_label() == 'box A'
     assert all(x.get_label().startswith("_") for x in bp4['medians'][1:])
+
+
+def test_patchcollection_legend():
+    # Test that PatchCollection labels show up in legend (issue #23998)
+    fig, ax = plt.subplots()
+
+    # Create some patches
+    p1 = mpatches.Polygon([[0, 0], [100, 100], [200, 0]])
+    p2 = mpatches.Polygon([[400, 0], [500, 100], [600, 0]])
+
+    # Create a PatchCollection with a label
+    pc = mcollections.PatchCollection([p1, p2], label="patch collection",
+                                      facecolor='blue', edgecolor='black')
+    ax.add_collection(pc, autolim=True)
+    ax.autoscale_view()
+
+    # Create the legend
+    leg = ax.legend()
+
+    # Check that the legend contains our label
+    assert len(leg.get_texts()) == 1
+    assert leg.get_texts()[0].get_text() == "patch collection"
+
+    # Check that there's a legend handle
+    handles = leg.legend_handles
+    assert len(handles) == 1
+    assert isinstance(handles[0], mpatches.Rectangle)
+
+    plt.close(fig)
+
+
+def test_patchcollection_legend_properties():
+    # Test that PatchCollection legend preserves visual properties
+    fig, ax = plt.subplots()
+
+    # Create patches with specific properties
+    p1 = mpatches.Circle((0, 0), 1)
+    p2 = mpatches.Circle((2, 0), 1)
+
+    # Create a PatchCollection with specific visual properties
+    pc = mcollections.PatchCollection(
+        [p1, p2],
+        label="styled patches",
+        facecolor='red',
+        edgecolor='blue',
+        linewidths=3,
+        linestyle='--'
+    )
+    ax.add_collection(pc)
+    ax.autoscale_view()
+
+    # Create the legend
+    leg = ax.legend()
+    legend_patch = leg.legend_handles[0]
+
+    # Verify that visual properties are preserved
+    assert_allclose(legend_patch.get_facecolor()[:3],
+                    pc.get_facecolor()[0][:3], rtol=1e-5)
+    assert_allclose(legend_patch.get_edgecolor()[:3],
+                    pc.get_edgecolor()[0][:3], rtol=1e-5)
+    assert legend_patch.get_linewidth() == pc.get_linewidths()[0]
+    assert legend_patch.get_linestyle() == pc.get_linestyles()[0]
+
+    plt.close(fig)
+
+
+def test_patchcollection_legend_match_original():
+    # Test PatchCollection legend with match_original=True
```

Comment:
Do we need this test? The legend patch takes the properties from the first patch. `match_original` applies the properties from the first patch to all other patches. I don't see how any reasonable implementation of `match_original` could adversingly affect the legend.

Reply:
I've removed the `test_patchcollection_legend_match_original` test. You're absolutely right - since the legend handler always takes properties from the first patch in the collection, `match_original` doesn't affect the legend behavior. The test was redundant with `test_patchcollection_legend_properties`.

Now we have 3 focused tests:
1. `test_patchcollection_legend` - Basic functionality (label appears in legend)
2. `test_patchcollection_legend_properties` - Visual properties are preserved
3. `test_patchcollection_legend_empty` - Edge case: empty collection doesn't crash